src/srv.rs

```diff
@@ -103,15 +109,19 @@ impl SrvResolver {
             }
 
             if !&hostname_parts[1..].ends_with(domain_name) {
-                srv_addresses.push(Err(ErrorKind::DnsResolve {
-                    message: format!(
-                        "SRV lookup for {} returned result {}, which does not match domain name {}",
-                        original_hostname,
-                        address,
-                        domain_name.join(".")
-                    ),
+                if matches!(dm, DomainMismatch::Error) {
+                    return Err(ErrorKind::DnsResolve {
+                        message: format!(
+                            "SRV lookup for {} returned result {}, which does not match domain \
+                             name {}",
+                            original_hostname,
+                            address,
+                            domain_name.join(".")
+                        ),
+                    }
+                    .into());
                 }
-                .into()));
+                continue;
```

Comment:
Previously there was no `continue` here, which was kind of a nasty bug-in-waiting - invalid hosts would still have been added to the list.

Reply:
good catch!